### DISCUSSION TITLE
Release 3.0.0 - Turing 5" display support!

### DISCUSSION THREAD

**Discussion by mathoudebine** [mathoudebine/turing-smart-screen-python#256] [Announcements]
<img src="https://github.com/mathoudebine/turing-smart-screen-python/assets/38615348/68c7db1c-f746-4ea5-bf83-913a5239b1cf" width="600" />

This release comes with Turing 5" displays support! 
Many thanks to @alexwbaule @arthurferrai @MrFr1day and everyone that contributed in issues #81 and #90 
Please note there is only basic support for now: no video or storage.

Documentation, Theme Editor, Configuration Wizard and Github workflows have been updated to handle 5" devices as well.  
This release comes without any breaking changes for 3.5" devices.  

### Configuration Wizard
In the configuration Wizard, you can now select Turing 5" as your smart screen model.  
<img src="https://github.com/mathoudebine/turing-smart-screen-python/assets/38615348/c6db8368-8abd-4d2d-8516-64fabb9a64e9" height="150" />

### Themes
Because the display resolution is different for 3.5" displays (320x480) and 5" displays (800x480), themes made previously for 3.5" displays cannot be used on 5" displays.  
New 5" themes have been developed and packaged in this release:

<img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/5inchTheme2/preview.png" width="150" /> <img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/CyanTheme/preview.png" width="150" /> <img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/DragonBall5inch/preview.png" height="150" /> <img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/Landscape15Grid/preview.png" height="150" /> <img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/NZXT_B/preview.png" height="150" /> <img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/NZXT_BLUR/preview.png" height="150" /> <img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/NZXT_C/preview.png" height="150" /> <img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/NZXT_color/preview.png" height="150" /> <img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/NZXT_dynamic/preview.png" height="150" /> <img src="https://raw.githubusercontent.com/mathoudebine/turing-smart-screen-python/main/res/themes/PurpleTheme/preview.png" height="150" />

When you select Turing 5" in the Configuration Wizard, the theme list is automatically updated to display only compatible themes. 
<img src="https://github.com/mathoudebine/turing-smart-screen-python/assets/38615348/a440ff9f-f0cc-4083-b093-d4245ec0a879" height="180" /> 

### `theme.yaml`
To create a theme for 5" devices, the format is the same as the 3.5" themes. You can do it manually or using the Theme Editor.  
The only property to add to your `theme.yaml` is:
``` yaml
display:
  DISPLAY_SIZE: 5"
```
If this property is not in the theme file (to keep compatibility with old/community themes) or if its value is different than `5"` then the theme will be for 3.5 displays.

## What's Changed
* Support 5 inches devices by @alexwbaule & @arthurferrai in https://github.com/mathoudebine/turing-smart-screen-python/pull/252
* New themes for 5" displays by @mathoudebine in https://github.com/mathoudebine/turing-smart-screen-python/pull/253

## New Contributors
* @alexwbaule made their first contribution in https://github.com/mathoudebine/turing-smart-screen-python/pull/252
* @arthurferrai made their first contribution in https://github.com/mathoudebine/turing-smart-screen-python/pull/252

**Full Changelog**: https://github.com/mathoudebine/turing-smart-screen-python/compare/2.4.3...3.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mathoudebine/turing-smart-screen-python/releases/tag/3.0.0'>Release 3.0.0 - Turing 5" display support!</a>.</em>

**Comment by DJRoby19** [comment]
Updated version 3.0.0 all working on Linux
good job

**Comment by Rembo777** [comment]
is it possible to display fps and frametime graph strip?

**Reply by mathoudebine** [reply]
Hi, not for now because I don't know to to get these data from the GPU bu I'll look into it!

**Reply by Rembo777** [reply]
this is a very useful parameter

**Reply by ooemir23** [reply]
Have you been able to deal with the fps? It's very important to me. @mathoudebine Thx

**Reply by mathoudebine** [reply]
FPS has now been implemented to the latest 3.3.1 release.
Feel free to try it out and add the FPS framerate to your favorite theme!

Please note it is for now limited to Windows platforms and AMD GPUs

**Comment by Paradoxum** [comment]
> Hi, not for now because I don't know to to get these data from the GPU bu I'll look into it!

Start at RTSS and how it creates frametime graphs etc I'm not sure how it works but I think that's the best starting point you can't just pull that stuff from the GPU like a sensor.

**Comment by DJRoby19** [comment]
[mathoudebine](https://github.com/mathoudebine)
I have a problem after 3.0 update when I do from terminal linux mint python3 main.py this error comes out
2023-07-20 17:58:57 [ERROR] Cannot open COM port /dev/ttyACM1: [Errno 19] could not open port /dev/ttyACM1: [Errno 19] No device matches: '/dev/ttyACM1'
but then it loads the screenshot
on the old version I had no problem what can it be
